### PR TITLE
CA-38 Add dest-specific numeric / json prop handling

### PR DESCRIPTION
**Pull request by bucrogers**
# TL;DR
For those destinations which have props which the backend requires be numeric or json objects, this ensures they are sent that way.

# Summary
The destinations / actions affected are:
* Facebook cAPI
* theTradeDesk
  *  a json parse error is now handled as a warning rather than a do-not-send error
* track - the json processing now occurs based on prop name rather than leading [ or { (to be consistent with the dest-specific handling)

# Testing notes
Tested permutations of each of the following on the GTM container fptest--buc+test:
- [x] Facebook cAPI
- [x] theTradeDesk (retested with new warning behavior)
- [x] track (retested with new json behavior)

### CODE REVIEW

**Review by 626D6172 — APPROVED** [review]
LGTM with pair review
